#Scenes/SceneScripts/MainMenu.gd

```diff
@@ -1,20 +1,38 @@
 extends Control
 ## Control the flow from the main menu
 
+var continue_button: TextureButton
+
 #on opening the main menu we grab the focus of the start button
 func _ready() -> void:
 	$"Menu buttons/Continue".grab_focus()
 	$"game version".text = ProjectSettings.get_setting("application/config/version")
 	for child in get_node("Menu buttons").get_children():
-			child.mouse_entered.connect(_on_button_hovered.bind(child))
+		if child.name == "Continue":
+			continue_button = child
+		child.mouse_entered.connect(_on_button_hovered.bind(child))
+	if !PlayerManager.has_saved_data():
+		continue_button.disabled = true
+	
 
 ## What happens when the start button is pressed
 func _on_start_pressed() -> void:
-	if not PlayerManager.is_player_initial_position_set:
-		SceneManager.goto_scene("res://#Scenes/MapUI.tscn")
-	else:
-		# TODO : load the last scene the player was in
-		pass
+	SaveManager.clear_data()
+	PlayerManager.clear_data()
+	MapManager.clear_map_data()
+	XpManager.clear_data()
+	CardManager.clear_data()
```

Comment:
should they be named `init_data` instead of `clear_data` ? clear makes it sound like we are deleting things, when we are creating new things

Also why is the map the only one with a special name ?

Reply:
I can go either way, I'll rename it to init_data, and I just have bad naming convention sense.

Reply:
I think everyone has bad naming convention sense x)